src/ECDSASignatureChecker.sol

```diff
@@ -0,0 +1,170 @@
+     * disperser (represented by apk in the parameters),
+     * - subtracting the pubkeys of all the signers not in the quorum (nonSignerPubkeys) and storing 
+     * the output in apk to get aggregated pubkey of all operators that are part of quorum.
+     * - use this aggregated pubkey to verify the aggregated signature under BLS scheme.
+     * 
+     * @dev Before signature verification, the function verifies operator stake information.  This includes ensuring that the provided `referenceBlockNumber`
+     * is correct, i.e., ensure that the stake returned from the specified block number is recent enough and that the stake is either the most recent update
+     * for the total stake (of the operator) or latest before the referenceBlockNumber.
+     * @param msgHash is the hash being signed
+     * @param quorumNumbers is the bytes array of quorum numbers that are being signed for
+     * @param signerIds TODO: document
+     * @param signatures TODO: document
+     * @return quorumStakeTotals is the struct containing the total and signed stake for each quorum
+     * @return signatoryRecordHash is the hash of the signatory record, which is used for fraud proofs
+     */
+    function checkSignatures(
+        bytes32 msgHash, 
+        bytes calldata quorumNumbers,
+        bytes32[] memory signerIds,
+        bytes[] memory signatures
+    ) 
+        public 
+        view
+        returns (
+            QuorumStakeTotals memory,
+            bytes32
+        )
+    {
+        require(
+            signerIds.length == signatures.length,
+            "BLSSignatureChecker.checkSignatures: signature input length mismatch"
+        );
+        // For each quorum, we're also going to query the total stake for all registered operators
+        // at the referenceBlockNumber, and derive the stake held by signers by subtracting out
+        // stakes held by nonsigners.
+        QuorumStakeTotals memory stakeTotals;
+        stakeTotals.totalStakeForQuorum = new uint96[](quorumNumbers.length);
+        stakeTotals.signedStakeForQuorum = new uint96[](quorumNumbers.length);
+
+        // Get a bitmap of the quorums signing the message, and validate that
+        // quorumNumbers contains only unique, valid quorum numbers
+        uint256 signingQuorumBitmap = BitmapUtils.orderedBytesArrayToBitmap(quorumNumbers, registryCoordinator.quorumCount());
+
+        for (uint256 i = 0; i < signerIds.length; ++i) {
+
+            // The check below validates that operatorIds are sorted (and therefore free of duplicates)
+            if (i != 0) {
+                require(
+                    uint256(signerIds[i]) >  uint256(signerIds[i - 1]),
+                    "ECDSASignatureChecker.checkSignatures: signer keys not sorted"
+                );
+            }
+
+            // check the operator's signature
+            EIP1271SignatureUtils.checkSignature_EIP1271(address(uint160(uint256(signerIds[i]))), msgHash, signatures[i]);
```

Comment:
related to my previous comments, if we store the operatorId as operatorAddr instead (literally type `address` instead of `bytes32`), we don't need to do this conversion here (`address(uint160(uint256))`). Afaik there's no other use for storing the operatorId as a bytes32 instead of an address?

Reply:
Changed it in my branch and everything checked out.